framework/validators/DateValidator.php

```diff
@@ -266,6 +266,12 @@ public function init()
     public function validateAttribute($model, $attribute)
     {
         $value = $model->$attribute;
+
+        if ($this->skipOnEmpty === false && empty($value) && $this->timestampAttribute !== null) {
```

Comment:
if timestampAttribute is already NULL, a different logic will apply, continuing in the method. To what extent is this logical and comprehendable? Its a bit hard to say. Could you improve this on readability in order for it to be easily understood?